### PR TITLE
MAYA-105322 - No undo support for viewport selections of USD objects

### PR DESCRIPTION
**Pull request by seando-adsk**
#### MAYA-105322 - No undo support for viewport selections of USD objects
* Use new Ufe::NamedSelection "MayaSelectTool" for viewport selection, rather than modifying UFE global selection directly.

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp [hunk 1]
```diff
@@ -755,7 +758,9 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
     // each intersection.
 #if defined(MAYA_ENABLE_UPDATE_FOR_SELECTION)
     const TfToken&                 selectionKind = _selectionKind;
+#if UFE_PREVIEW_VERSION_NUM < 2027 // #ifndef UFE_V2_FEATURES_AVAILABLE
     const MGlobal::ListAdjustment& listAdjustment = _globalListAdjustment;
+#endif
```
Comment (seando-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp:763]: Don't this need var anymore with the new named selection code.

File: lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp [hunk 2]
```diff
@@ -783,6 +788,10 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
         return false;
     }
 
+#if UFE_PREVIEW_VERSION_NUM >= 2027 // #ifdef UFE_V2_FEATURES_AVAILABLE
+    auto ufeSel = Ufe::NamedSelection::get("MayaSelectTool");
+    ufeSel->append(si);
+#else
```
Comment (seando-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp:794]: We simply append any items to select to this named selection. Maya takes care of the replace/toggle/add/deselect.

File: lib/mayaUsd/ufe/UsdStageMap.cpp [hunk 3]
```diff
@@ -70,7 +70,7 @@ void UsdStageMap::addItem(const Ufe::Path& path, UsdStageWeakPtr stage)
 {
     // We expect a path to the proxy shape node, therefore a single segment.
     auto nbSegments =
-#ifdef UFE_V0_2_6_FEATURES_AVAILABLE
+#if UFE_PREVIEW_VERSION_NUM >= 2006
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/UsdStageMap.cpp:73]: Fixed this ifdef as I noticed it was wrong.

File: lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp (hunk 2 above)
Comment (seando-adsk) [reply lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp:794]: Same changes in all the plugins.

**Review by kxl-adsk — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Comment by HdC-adsk** [comment]
I like the notion of named selection. Is it intended to support multiple named selections?

**Review by HdC-adsk — APPROVED** [review]

**Comment by seando-adsk** [comment]
@HdC-adsk Yes, UFE supports any number of named selections. The first time you get one with a specific name it will be created. Subsequent calls to get with that name return the same one.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**